Sources/PackageModel/Target.swift

```diff
@@ -666,3 +668,12 @@ public enum PluginCapability: Equatable, Codable {
         }
     }
 }
+
+
+fileprivate extension AbsolutePath {
+  var isDirectory: Bool {
+      var isDir: ObjCBool = false
+      let exists = FileManager.default.fileExists(atPath: self.pathString, isDirectory: &isDir)
+      return exists && isDir.boolValue
```

Comment:
I think we need to go through the `FileSystem` abstraction here instead of checking the local file system directly. It might be a bit awkward to get a handle to the currently used file system object here, one option could be moving the check to `TargetSourcesBuilder`.

Reply:
Moved a check out of this